Kafka Stream Test Cases Implementation

obedjamir opened the pull request:
All classes coverage at 100%
Except the retry conditions in KafkaStreamConfig

obedjamir committed:
Kafka Stream Test Cases Implementation
All classes coverage at 100%
Except the retry conditions in KafkaStreamConfig

PinakiGhosh commented on service/src/main/java/com/bankbazaar/kafka/service/config/KafkaStreamsConfig.java:
```diff
@@ -61,29 +65,25 @@ public Function<KStream<String,Data>, KStream<String,Data>> fileProcessor(@Quali
                                     File file = new File(classLoader.getResource(".").getFile() + value.getFileName());
                                     if (file.exists()) {
-                                        fileStatusService.updateEntry(value,Status.FAILURE);
+                                        this.kafkaTemplate.send(TOPIC,fileStatusService.updateEntry(value,Status.FAILURE));
```
Why do we need to add code to send to topic? Doesnt Kafka streams take care of it on ts own?

obedjamir replied:
Branced stream has been implemented.

PinakiGhosh commented on service/src/main/java/com/bankbazaar/kafka/service/config/KafkaStreamsConfig.java:
```diff
@@ -61,29 +65,25 @@ public Function<KStream<String,Data>, KStream<String,Data>> fileProcessor(@Quali
-                                fileData.setId(value.getId());
-                                fileData.setStatus(Status.ERROR);
-                                fileStatusService.update(fileData);
+                                this.kafkaTemplate.send(TOPIC,fileStatusService.updateEntry(value,Status.ERROR));
```
This should not be required. Configure the stream to handle it automatically

PinakiGhosh commented on service/src/main/java/com/bankbazaar/kafka/service/service/FileStatusService.java:
```diff
@@ -36,11 +33,17 @@ public FileStatusEntity createEntry(DataDto data)
+    {
+        return fileStatusManager.getEntry(id).get();
+    }
+    public FileStatusEntity updateEntry(Data data, Status status)
```
Take only the dataId and status as input.
The function should be like
1. Find entity by Id
2. Set status
3. Save

The main idea behind this is since there are multiple functions which want to update the status why do all the functions send entire objects? Only the id and new status should be enough.

Also when I know I am only going to update the status why do I need a full update function?

obedjamir replied:
Update entry has been modified to take only id and status as arguments.

obedjamir committed:
Branced Stream Implementation
Used .branch() to conditionally direct to different topics
Also, modified update entry to take only id and status as arguments

obedjamir committed:
Branced Stream Implementation
Conditional branch to seperate topics
